server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader12to13.java

```diff
@@ -64,6 +70,49 @@ public void upgradeMetadata(ServerContext context) {
     addHostingGoalToUserTables(context);
     deleteExternalCompactionFinalStates(context);
     deleteExternalCompactions(context);
+    removeCompactColumns(context);
+  }
+
+  private void removeCompactColumns(ServerContext context) {
+
+    final Text COMPACT_QUAL = new Text("compact");
+
+    try (var scanner = context.createScanner(MetadataTable.NAME);
+        var writer = context.createBatchWriter(MetadataTable.NAME)) {
+      scanner.setRange(MetadataSchema.TabletsSection.getRange());
+
+      for (Map.Entry<Key,Value> entry : scanner) {
+        var key = entry.getKey();
+        var row = key.getRow();
+        Preconditions.checkState(key.getColumnVisibilityData().length() == 0,
```

Comment:
While this should always be true - why are you checking this here?  Or, should I add this check to the 11 to 12 upgrade code?